### ISSUE TITLE
Editing in files that are blocked because of being in the dependency chain does not automatically unblock them

### ISSUE THREAD

**Issue by RalfJung** [psteckler/ProofGeneral#75]
Say file `B` imports `A`, and I have the interactive focus in `B`. If I now open `A`, PG shows the entire file as red because things are "locked", which makes sense -- after all, changes here could break `B`. However, when I try to type in that red region, PG master automatically offers to change the interactive focus to `A` -- while this branch of PG just refuses to change the text. I have to manually C-c C-RET, confirm, C-c C-RET. It would be nice if the behavior from PG master could be adapted here (actually I am not even sure why it is different, but I guess this is not part of the code that can be shared).

**Comment by psteckler** [comment]
I'm actually unsure about how to get a file to appear in red in PG. When I open a new file, PG offers to retract the current script.

What's a small example?

**Comment by RalfJung** [comment]
I've had the dependent-on file open already.

```
$ cat a.v
Definition foo := True.
$ cat b.v
Require Import a.
```

* Open both a.v and b.c in emacs
* In b.v, do C-c C-b
* Switch to a.b to see it in "all red, not editable" state

**Comment by psteckler** [comment]
Hmm, I'm not seeing the all-red locked state in a.v. It's freely editable.

Could there be a setting in PG that enables the locking? I was not able to find it, if so.

**Comment by RalfJung** [comment]
I did some investigation, and it seems that the locking happens only when I have `coq-compile-before-require` enabled (which doesn't seem to be the default, strange enough).

**Comment by psteckler** [comment]
Behavior confirmed, investigating.

**Comment by psteckler** [comment]
I've committed a fix. Keep an eye out for anything I may have broken with the "sent span", the light blue part of the buffer.

**Comment by RalfJung** [comment]
Confirmed. Thanks!
